sky/serve/service.py

```diff
@@ -83,6 +83,28 @@ def _cleanup_storage(task_yaml: str) -> bool:
     return True
 
 
+def _cleanup_storage_from_service_name(service_name: str,
+                                       task_yaml: str) -> bool:
+    """Clean up the storage from service_name.
+
+    Args:
+        service_name: The service name.
+        task_yaml: The task yaml file.
+
+    Returns:
+        True if the storage is cleaned up successfully, False otherwise.
+    """
+    # TODO(MaoZiming): storage from multiple versions might
+    # need to be cleaned up
+    version = serve_state.get_service_version(service_name)
+    if version is not None:
+        task_yaml = serve_utils.generate_task_yaml_file_name(
+            service_name, version)
+        logger.info(f'Cleaning up storage for service {service_name} '
+                    f'with version {version} and task yaml {task_yaml}')
+    return cleanup_storage(task_yaml)
```

Comment:
Should we clean up any remaining versions instead of only the latest version? Reason: to make our system more robust to bugs

Reply:
These old versions will be cleaned when the service is updated and there is no replica of that old version.
https://github.com/skypilot-org/skypilot/blob/serve-update-new/sky/serve/replica_managers.py#L777-L778

Reply:
If our user `sky serve down` in the process of a rolling update (say we still have 2 old ready replica and only one latest ready replica), it is possible that when we run this function, there is more than one version?

Reply:
fixed.